src/core/Akka/Actor/Props.cs

```diff
@@ -285,37 +155,51 @@ public Props(Deploy deploy, Type type, IEnumerable<object> args)
         }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="Props" /> class.
+        ///     Initializes a new instance of the <see cref="Props" /> class.
         /// </summary>
         /// <param name="deploy">The configuration used to deploy the actor.</param>
         /// <param name="type">The type of the actor to create.</param>
         /// <param name="args">The arguments needed to create the actor.</param>
-        /// <exception cref="ArgumentException">This exception is thrown if <paramref name="type"/> is an unknown actor producer.</exception>
-        public Props(Deploy deploy, Type type, params object[] args)
+        /// <exception cref="ArgumentException">This exception is thrown if <paramref name="type" /> is an unknown actor producer.</exception>
+        public Props(Deploy deploy, Type type, params object[] args) 
+            : this(CreateProducer(type, args), deploy, args) // have to preserve the "CreateProducer" call here to preserve backwards compat with Akka.DI.Core
```

Comment:
First big change - this used to be the "bottom" constructor that all other constructors call. I've added one more `internal` one below it. 

We needed to keep the mysterious / bad `CreateProducer` call here in order not to completely blow up Akka.DI.Core support, which many users are probably still using.